paddle/fluid/platform/device_context.cc

```diff
@@ -271,8 +310,14 @@ CUDADeviceContext::~CUDADeviceContext() {
 Place CUDADeviceContext::GetPlace() const { return place_; }
 
 void CUDADeviceContext::Wait() const {
+  auto& allocator =
+      DeviceTemporaryAllocator::Instance().Get<CUDADeviceContext>(*this);
+  allocator.MoveToDeleteQueue();
```

Comment:
Do we really need 2 methods `MoveToDeleteQueue` and `Release`. It seems that they would not be invoked separately.

Reply:
`MoveToDeleteQueue`  is to move the temp_allocations that has not been held by any variable to delete_queue before call `cudaStreamSynchronize`,  `Release` is to delete the  allocation of delete_queue.
This is because, if we call the `Release` after `cudaStreamSynchronize`,  this will lead to some allocations are released before calling `cudaStreamSynchronize`.